sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala

```diff
@@ -490,6 +491,7 @@ class LocalityTestFileSystem extends RawLocalFileSystem {
 
   override def getFileBlockLocations(
       file: FileStatus, start: Long, len: Long): Array[BlockLocation] = {
+    require(!file.isDirectory, "The file path can not be a directory.")
```

Comment:
This test suite already have test cases that create multiple partitions in a table. and the code path in this PR is visited by these test cases. Adding the require here will catch any case where a directory is passed to generate block locations.